src/engine/Util/Detector.ts

```diff
@@ -5,9 +5,117 @@ module ex {
     * Excalibur internal feature detection helper class
     */
    export class Detector {
-      
+
+      private _features: Object = null;
+
       public failedTests: string[] = [];
+
+      public constructor() {
+         this._features = this._loadBrowserFeatures();
+      }
+
+      /**
+       * Returns a map of currently supported browser features. This method
+       * treats the features as a singleton and will only calculate feature
+       * support if it has not previously been done.
+       * 
+       * @return {Object} the list of supported features
+       */
+      public getBrowserFeatures(): Object {
+         if (this._features === null) {
+            this._features = this._loadBrowserFeatures();
+         }
+         return this._features;
+      }
       
+      /**
+       * Report on non-critical browser support for debugging purposes.
+       * Use native browser console colors for visibility.
+       * 
+       * @return {void}
+       */
+      public logBrowserFeatures(): void {
+         let features = {
```

Comment:
Would it make sense to pull this up out of the class towards the top of the file so we can see this as a reminder to update if we add new features later on?